test/collect.cc

```diff
@@ -39,5 +39,35 @@ TEST(Collect, CommonSetPass) {
   EXPECT_EQ(12, *++result.begin());
 }
 
+TEST(Collect, VectorToRepeatedPtrField) {
+  std::vector<std::string> v = {"Hello", "World"};
+
+  auto s = [&]() {
+    return Iterate(v)
+        | Collect<google::protobuf::RepeatedPtrField<std::string>>();
+  };
+
+  google::protobuf::RepeatedPtrField<std::string> result = *s();
+
+  EXPECT_EQ(result.size(), 2);
```

Comment:
Prefer `ASSERT` over `EXPECT` when checking a container's size before dereferencing elements, or doing other things that could crash if previous `EXPECT` statements fail.

> Use ASSERT when it doesn't make sense to continue

https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html